base/habana-python-3.11/Dockerfile

```diff
@@ -0,0 +1,82 @@
+# Reference:
+#  - https://vault.habana.ai/ui/repos/tree/General/gaudi-docker/1.17.1/rhel9.4/habanalabs/pytorch-installer-2.3.1/1.17.1-40
+FROM vault.habana.ai/gaudi-docker/1.17.1/rhel9.4/habanalabs/pytorch-installer-2.3.1:1.17.1-40 as base
```

Comment:
For upstream, we are okay to take these images build out of RH. But in downstream (and after the move to konflux eventually), I'm not sure whether we should use these. Thus saying it now so we don't realize that we will need a different approach in our downstream repository.

Truth is that since the [Dockerfile](https://github.com/HabanaAI/Setup_and_Install/blob/main/dockerfiles/pytorch/Dockerfile.rhel9.4) for this is available, then we can still use this approach - just build even this source image ourselves for the downstream/product build.

Sidenote: also looks like the download of that image from the vault.habana.ai isn't very fast - it took me 24 minutes to download it using the internet connection in the office :upside_down_face: 

---

Another question - do we want to use the pytorch flavor as a base image here or do we want to start our build on top of a clean habana image - https://github.com/HabanaAI/Setup_and_Install/blob/main/dockerfiles/base/Dockerfile.rhel9.4 ?

Reply:
It does make sense and I'm thinking about a way to make that change look good. I will think a little bit about the change and will add a comment here later.

Reply:
I did download the Dockerfile from Habana's repository into the `base/` folder, added one more step in our process to build the three images accordingly:

```
# Build and push all 3 images related to Habana 1.17.1:
#   - base-habana-python-3.11-base
#   - base-habana-python-3.11
#   - habana-jupyter-1.17.1-ubi9-python-3.11
#
.PHONY: base-habana-1.17.1-python-3.11-base
base-habana-1.17.1-python-3.11-base:
	$(call image,$@,base/habana-1.17.1-python-3.11-base)

.PHONY: base-habana-1.17.1-python-3.11
base-habana-1.17.1-python-3.11: base-habana-1.17.1-python-3.11-base
	$(call image,$@,base/habana-1.17.1-python-3.11)

.PHONY: habana-jupyter-1.17.1-ubi9-python-3.11
habana-jupyter-1.17.1-ubi9-python-3.11: base-habana-1.17.1-python-3.11
	$(call image,$@,habana/1.17.1/ubi9-python-3.11,$<)
```

This created a dependency across our images, in a way that:

```mermaid
%%{init: {"flowchart": {"htmlLabels": false}} }%%
flowchart LR
    base1["base-habana-python-3.11-base"]
    base2["base-habana-python-3.11"]
    base3["habana-jupyter-1.17.1-ubi9-python-3.11"]
    base1 --> base2
    base2 --> base3
```

PS: names on the chart have been simplified just for viewing purposes

Reply:
Thank you for this change, Daniel! This approach seems better to me since we avoid dependency on their vault, though we'll need to be sure we are periodically in sync with the changes in that repo in case we need to.

I'm not sure I like the `base-*-base` naming here. Though, I'm not able to come up with anything better myself. Maybe something like `base-*-origin`? Or `base-*-intel`? Maybe others will be able to think of something better.

Anyway, I would like also others to provide their point of view on this approach, whether we're not missing something more here.